Updated the code to match the new opac.sbn.it site

doriansoru opened the pull request:
The translator for opac.sbn.it does not work anymore, because the website was updated.

I tried to update it.

Dorian

AbeJellinek commented on sbn.it.js:
```diff
@@ -1,15 +1,15 @@
-	"creator": "Philipp Zumstein",
-	"target": "^https?://(www|opac)\\.sbn\\.it/opacsbn/opaclib",
+	"creator": "Philipp Zumstein, Dorian Soru",
+	"target": "^https?:\\/\\/(www|opac)\\.sbn\\.it\\/(web\\/)?(opacsbn\\/)?risultati-ricerca-avanzata",
```
Don't need to escape slashes

AbeJellinek commented on sbn.it.js:
```diff
@@ -51,67 +56,130 @@ var typeMapping = {
+// A single item per page but pages change dynamically with scrolling
+function getDetailedResults(doc, url, checkOnly) {
+	// Detailed results
+	var parser = new DOMParser();
+	var innerdoc = parser.parseFromString(doc.documentElement.outerHTML, "text/html");
+	var results = ZU.xpath(innerdoc, '//div[@data-to-post-item]/@data-to-post-item');
```
We don't need DOMParser or XPath at all here, I think...

```suggestion
	var results = doc.querySelectorAll('[data-to-post-item]');
```

AbeJellinek commented on sbn.it.js:
```diff
@@ -51,67 +56,130 @@ var typeMapping = {
-		return "multiple";
+	var items = {};
+	for (var i = 0; i < results.length; i++) {
+		var href = detailUrl + JSON.parse(results[i].value).id;
+		var title = JSON.parse(results[i].value).title;
```
And then

```suggestion
		var href = detailUrl + JSON.parse(results[i].getAttribute('data-to-post-item')).id;
		var title = JSON.parse(results[i].getAttribute('data-to-post-item')).title;
```

AbeJellinek commented on sbn.it.js:
```diff
@@ -51,67 +56,130 @@ var typeMapping = {
 
+// Get multiple results. The page is updated dynamically
 function getSearchResults(doc, checkOnly) {
+	var parser = new DOMParser();
+	var innerdoc = parser.parseFromString(doc.documentElement.outerHTML, "text/html");
+	var results = ZU.xpath(innerdoc, "//input[@data-to-post-item]/@data-to-post-item");
```
This doesn't need to be a separate method from the above

AbeJellinek commented on sbn.it.js:
```diff
@@ -51,67 +56,130 @@ var typeMapping = {
+			if (result) {
+				results = getDetailedResults(doc, url, false);
+				Zotero.selectItems(results, function (items) {
+					if (!items) {
+						return;
+					}
+					var ids = [];
+					for (var i in items) {
+						ids.push(i);
+					}
+					ZU.processDocuments(ids, scrape);
```
```suggestion
					if (items) ZU.processDocuments(Object.keys(items), scrape);
```

AbeJellinek commented on sbn.it.js:
```diff
@@ -51,67 +56,130 @@ var typeMapping = {
+			translator.translate();
+		});
 	});
 }/** BEGIN TEST CASES **/
```
Two newlines before comment

AbeJellinek commented on sbn.it.js:
```diff
@@ -140,46 +208,12 @@ var testCases = [
-			}
-		]
+		"url": "https://opac.sbn.it/risultati-ricerca-avanzata/-/opac-adv/all?fieldstruct%5B1%5D=ricerca.parole_tutte%3A4%3D6&fieldvalue%5B1%5D=sant%27agostino&fieldaccess%5B1%5D=Any%3A1016%3Anocheck&struct%3A1001=ricerca.parole_almeno_una%3A%40or%40",
+		"items": "multiple"
```
Are there no non-multiple pages anymore?

AbeJellinek reviewed (changes requested):
Thanks! Some comments.

doriansoru commented:
Thanks Abe,

I updated the code with your suggestions. Meanwhile the website seemed to be modified again and, in particular, I found an update on a single resource url. So I made to additional changes:

- updated regexId (l. 39 of the attachment)
- if (url.includes("detail")) -> if (url.includes("detail") || url.includes("index"))
- var id = _url.match(regexId)[4]; (it was [2])
- Simplified the scrape function

see the commit
